cmd/cli/proxy_configdump.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"helm.sh/helm/v3/pkg/action"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+)
+
+const dumpConfigDescription = `
+This command will dump the sidecar proxy configuration for the given pod.
+`
+
+const dumpConfigExample = `
+# Dump the proxy configuration for pod 'bookbuyer-5ccf77f46d-rc5mg' in the 'bookbuyer' namespace
+osm proxy dump-config bookbuyer-5ccf77f46d-rc5mg -n bookbuyer
+`
+
+type proxyDumpConfigCmd struct {
+	out        io.Writer
+	config     *rest.Config
+	clientSet  kubernetes.Interface
+	namespace  string
+	pod        string
+	localPort  uint16
+	sigintChan chan os.Signal
+}
+
+func newProxyDumpConfig(config *action.Configuration, out io.Writer) *cobra.Command {
+	dumpConfigCmd := &proxyDumpConfigCmd{
+		out:        out,
+		sigintChan: make(chan os.Signal, 1),
+	}
+
+	cmd := &cobra.Command{
+		Use:   "dump-config POD ...",
+		Short: "dump proxy config",
+		Long:  dumpConfigDescription,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			dumpConfigCmd.pod = args[0]
+			conf, err := config.RESTClientGetter.ToRESTConfig()
+			if err != nil {
+				return errors.Errorf("Error fetching kubeconfig")
+			}
+			dumpConfigCmd.config = conf
+
+			clientset, err := kubernetes.NewForConfig(conf)
+			if err != nil {
+				return errors.Errorf("Could not access Kubernetes cluster. Check kubeconfig")
+			}
+			dumpConfigCmd.clientSet = clientset
+			return dumpConfigCmd.run()
+		},
+		Example: dumpConfigExample,
+	}
+
+	//add mesh name flag
+	f := cmd.Flags()
+	f.StringVarP(&dumpConfigCmd.namespace, "namespace", "n", metav1.NamespaceDefault, "Namespace of pod")
+	f.Uint16VarP(&dumpConfigCmd.localPort, "local-port", "p", constants.EnvoyAdminPort, "Local port to use for port forwarding")
+
+	return cmd
+}
+
+func (cmd *proxyDumpConfigCmd) run() error {
+	// Check if the pod belongs to the mesh
+	pod, err := cmd.clientSet.CoreV1().Pods(cmd.namespace).Get(context.TODO(), cmd.pod, metav1.GetOptions{})
+	if err != nil {
+		return errors.Errorf("Could not find pod %s in namespace %s", cmd.pod, cmd.namespace)
+	}
+	if !isMeshedPod(*pod) {
```

Comment:
This is really nitpicking, but the comment here says `the mesh` whereas the comments in `isMeshedPod` says `a mesh`, so does it matter which mesh the pod belongs to?

Reply:
Changing to `a mesh` since we don't care which specific mesh the pod belongs to.